### ISSUE TITLE
[All] fix some programing errors reported by CppCheck

### ISSUE THREAD

**Issue by jotac0** [raysan5/raylib#1004]
Please, before submitting a new issue verify and check:

 - [x] I tested it on latest raylib version from master branch
 - [x] I checked there is no similar issue already reported
 - [ ] My code has no errors or misuse of raylib (it has, that is what this commit is about, preventing them :P)

### Issue description

I decided to run CppCheck, a static code analysis. and found a few issues.
Statistics
    Errors:    13
    Warnings:    32
    Style warnings:    636
    Portability warnings:    3
    Performance warnings:    0
    Information messages:    0

some of the errors are in raylib, others in the dependencies.
Errors that i fixed ( will be in separate pull request):

in models.c function LoadModelAnimations  we could close 2 times the file, and even try to read a closed file.

in physac.h in functions DestroyPhysicsBody and DestroyPhysicsManifold there were errors where we could access the index -1 of an array

in audio.c function InitAudioBuffer we were first using a created variable, only after checking if it was created

in rlgl.h if the function rlLoadTexture  was called with NULL pointer, we would dereference it

in shapes.c function GetShapesTexture we were comparing an unsigned value if it was equal or less than 0.

in text.c function TextToUtf8 we were not using the new pointer

and the most noticeable problem:

in textures.c function GetImageDataNormalized there was a missing break.

minor details:

we have in many places, variables that we create with an initial value, but is never used.

portability issues:
in dr_mp3 we have 2 shifts on negative values, this is undefined behavior...

in jar_xm.h we have a direct cast of pointers (integer* to float*).

**Comment by jotac0** [comment]
#1005 has the fixes :)

**Comment by raysan5** [comment]
@jotac0 thank you very much for the detailed review and the improvements submitted.

**Comment by jotac0** [comment]
just a note: i did found some false positives, and there is more errors there that should not be a concern (like failing to start glad in case of an error, and reporting it working, this would probably lead to instant crash!)

**Comment by raysan5** [comment]
@jotac0 ok, no problem, I'm reviewing and merging your changes... are you adding some PR?

**Comment by jotac0** [comment]
@raysan5 the other problems i found, are a lot harder to understand.

for example:
at rlgl.h:3432
the program says "Finding variables 'mipWidth' and 'mipHeight' that are assigned the same expression is suspicious and might indicate a cut and paste or logic error. Please examine this code carefully to determine if it is correct."

i have no idea if this is an issue or not.

**Comment by jotac0** [comment]
i will output a full error list, after my changes.

**Comment by jotac0** [comment]
suggestion: go to 
https://portableapps.com/apps/development/cppcheck-portable

and run that on the solution result of the cmake tool and see yourself
closing.
